packages/functions/src/dataTriggers.ts

```diff
@@ -461,3 +466,116 @@ export const createAttendedSlotOnAttendance = functions
 
     await batch.commit();
   });
+
+export const createCustomerStats = functions
+  .region(__functionsZone__)
+  .firestore.document(
+    `${Collection.Organizations}/{organization}/${OrgSubCollection.Bookings}/{secretKey}/${BookingSubCollection.BookedSlots}/{bookingId}`
+  )
+  .onWrite(async (_, context) => {
+    const { organization, secretKey } = context.params as Record<
+      string,
+      string
+    >;
+
+    const db = admin.firestore();
+
+    const bookingRef = db
+      .collection(Collection.Organizations)
+      .doc(organization)
+      .collection(OrgSubCollection.Bookings)
+      .doc(secretKey);
+
+    // Fetch the booking document
+    const bookingDoc = await bookingRef.get();
+    const { id: customerId } = bookingDoc.data() as CustomerBookings;
+
+    // Fetch documents from a subcollection of the booking
+    const bookedSlotsRef = bookingRef.collection("bookedSlots");
+    const bookedSlotsSnapshot = await bookedSlotsRef.get();
+
+    const bookedSlots: DocumentData = {};
+    bookedSlotsSnapshot.forEach((doc) => {
+      Object.assign(bookedSlots, { [doc.id]: doc.data() });
+    });
+
+    if (!Object.keys(bookedSlots).length) return;
+
+    // Get calendar day
+    const currentMonthStr = DateTime.now().toISODate().substring(0, 7);
+    const nextMonthStr = DateTime.now()
+      .plus({ month: 1 })
+      .toISODate()
+      .substring(0, 7);
+
+    const currentMonthSlots = (
+      await db
+        .collection(Collection.Organizations)
+        .doc(organization)
+        .collection(OrgSubCollection.SlotsByDay)
+        .doc(currentMonthStr)
+        .get()
+    ).data() as SlotsByDay;
+
+    const nextMonthSlots = (
+      await db
+        .collection(Collection.Organizations)
+        .doc(organization)
+        .collection(OrgSubCollection.SlotsByDay)
+        .doc(nextMonthStr)
+        .get()
+    ).data() as SlotsByDay;
+
+    const stats = Object.entries(bookedSlots).reduce(
```

Comment:
I see a great opportunity to reduce the size of this function (that starts online 475 and is passed to `onwrite`) by extracting the `stats` calculation outside.

To expand: this function currently takes care of two things. It fetches the data it needs and then combines it to the shape it needs. I'm suggesting to extract the second task into a named function, with its own (small) documentation. The new function should receive `currentMonthSlots` and `nextMonthSlots` and return a `stats` object. This new function will be testable without emulators.